ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/CommandParser.java

```diff
@@ -221,167 +210,123 @@ public static Object toFieldType(final Expression expressionValue) {
   */
   // CHECKSTYLE_RULES.OFF: CyclomaticComplexity
   public static SqlCommand transformToSqlCommand(
-      // CHECKSTYLE_RULES.ON: CyclomaticComplexity
-      final String sql, final Map<String, String> variables) {
+          // CHECKSTYLE_RULES.ON: CyclomaticComplexity
+          final String sql, final Map<String, String> variables) {
 
-    // Splits the sql string into tokens(uppercased keyowords, identifiers and strings)
-    final List<String> tokens = Arrays
-        .stream(sql.toUpperCase().split(QUOTED_STRING_OR_WHITESPACE))
-        .filter(s -> !s.isEmpty())
-        .collect(Collectors.toList());
-    switch (getStatementType(tokens)) {
+    validateSupportedStatementType(sql);
+    final String substituted;
+    try {
+      substituted = VariableSubstitutor.substitute(KSQL_PARSER.parse(sql).get(0), variables);
+    } catch (ParseFailedException e) {
+      throw new MigrationException(String.format(
+          "Failed to parse the statement. Statement: %s. Reason: %s",
+          sql, e.getMessage()));
+    }
+    final Class<? extends SqlBaseParser.StatementContext> statementContext =
```

Comment:
We should save the result of parsing the statement here so that we don't have to run the parser again for all the `getInsertValuesStatement`/`getCreateConnectorStatement` etc functions.